R/map_fetch.R

```diff
@@ -0,0 +1,268 @@
+#' @title Fetch aggregated density maps of GBIF occurrences
+#'
+#' @export
+#'
+#' @description This function is a wrapper for the GBIF mapping api version 2.0.
+#' The mapping API is a web map tile service making it straightforward to 
+#' visualize GBIF content on interactive maps, and overlay content from other 
+#' sources. It returns tile maps or vector maps with number of 
+#' GBIF records per area unit that can be used in a variety of ways, for example
+#' in interactive leaflet web maps. Map details are specified by a number of 
+#' query parameters, some of them optional. Full documentation of the GBIF
+#' mapping api can be found at https://www.gbif.org/developer/maps
+#'
+#' @param source (character): Either density for fast, precalculated tiles, 
+#' or adhoc for any search. (\code{default = "density"})
+#' 
+#' @param x (integer): the zoom. (\code{default = 0})
+#' 
+#' @param y (integer): the column. (\code{default = 0})
+#' 
+#' @param z (integer): the row. (\code{default = 0})
+#' 
+#' @param format (character): 
+#' .mvt for a vector tile
+#' /@Hx.png for a 256px raster tile (for legacy clients)
+#' /@1x.png for a 512px raster tile
+#' /@2x.png for a 1024px raster tile
+#' /@3x.png for a 2048px raster tile
+#' /@4x.png for a 4096px raster tile
+#' 
+#' @param srs (character):
+#' 	Spatial reference system. One of:
+#' 	EPSG:3857 (Web Mercator)
+#' 	EPSG:4326 (WGS84 plate care�)
+#' 	EPSG:3575 (Arctic LAEA)
+#' 	EPSG:3031 (Antarctic stereographic)
+#' 	see below under Projections.
+#' 	
+#' 	@param bin (optional): square or hex to aggregate occurrence counts into 
+#' 	squares or hexagons. Points by default.
+#' 	
+#' 	@param hexPerTile (optional): sets the size of the hexagons 
+#' 	(the number horizontally across a tile)
+#' 	
+#' 	@param squareSize (optional): sets the size of the squares. Choose a factor 
+#' 	of 4096 so they tessalate correctly: probably from 8, 16, 32, 64, 128, 
+#' 	256, 512.
+#' 	
+#' 	@param style (optional): for raster tiles, choose from the available styles. 
+#' 	Defaults to classic.point.
+#' 	
+#' 	@param search (optional): defines what type of subset of all GBIF data to 
+#' 	return. Should be one of c("taxonKey", "datasetKey", "country", "publisher",
+#' 	"publishingCountry"). Without any search parameter, all occurrences will be 
+#' 	returned.
+#' 	
+#' 	@param id (optional): defines the value to be used as filter criterium in 
+#' 	the category supplied by (\code{search}). Appropriate values depend on the
+#' 	search category that is used, for example integer for 
+#' 	(\code{search = "taxonKey"}). Has to be provided if 
+#' 	(\code{search}) parameter is specified.
+#' 	
+#' 	@param year (optional): limits the occurrences within a range of years, 
+#' 	for example c(2008, 2009, 2010).
+#' 	
+#' 	@param basisOfRecord (optional): can be given multiple times to include 

```

Comment:
make sure to say instead of given multiple times that the user passes in a character vector

Reply:
I've implemented this as an integer (or numeric) variable atm. Is that okay?

Reply:
`basisOfRecord` as an integer?

Reply:
Haha, no, I thought you were talking about year. Okay got it!